profiling/src/profile/mod.rs

```diff
@@ -509,27 +558,28 @@ impl From<&Profile> for pprof::Profile {
 
         if !profile.endpoints.mappings.is_empty() {
             for sample in samples.iter_mut() {
-                let mut endpoint: Option<&i64> = None;
-
-                for label in &sample.labels {
-                    if label.key == profile.endpoints.local_root_span_id_label {
-                        endpoint = profile.endpoints.mappings.get(&label.str);
-                        break;
+                // There _should_ only be one local root span id label, but it's not enforced.
+                let lrsi_labels: Vec<_> = sample
+                    .labels
+                    .iter()
+                    .filter(|label| label.key == profile.endpoints.local_root_span_id_label)
+                    .map(Clone::clone) // Need to clone to break borrows
+                    .collect();
+
+                for label in lrsi_labels {
```

Comment:
Minor: I somewhat wonder if adding a boolean to track that the label wasn't repeated in `Profile::add` is simpler than having to write extra code to handle the case it is here.